exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -478,28 +498,36 @@ public BatchGroup mergeAndSpill(LinkedList<BatchGroup> batchGroups) throws Schem
         break;
       }
       BatchGroup batch = batchGroups.pollLast();
+      assert batch != null;
```

Comment:
please add a message to the assertion, it will make our life easier if this assertion throws